ssl/ssl_locl.h

```diff
@@ -1947,6 +1949,13 @@ struct openssl_ssl_test_functions {
 
 const char *ssl_protocol_to_string(int version);
 
+/* Returns true if certificate and private key for 'idx' are present */
+static ossl_inline int ssl_has_cert(const SSL *s, int idx)
+{
+    return s->cert->pkeys[idx].x509 != NULL
```

Comment:
Well how it's used ensures idx is in range but there is no harm in adding a check: the compiler should optimise most of them away anyway.